lib/smooth/include/smooth/core/network/Wifi.h

```diff
@@ -89,5 +93,6 @@ namespace smooth::core::network
             std::string ssid{};
 
             std::string password{};
+            static ip4_addr_t ip;
```

Comment:
`ip` can now be accessed without ever creating an instance of `Wifi` which is the only one that initializes it meaning get_local_ip() can return uninitialized data.

`ip` must be initialized out-of-line in the top of Wifi.cpp: `ip4_addr_t Wifi;;ip = {0};`

Do we have threading issue here? `Wifi` is owned by the main task, but Wifi::get_local_ip()` may be called from any thread.

Reply:
I moved the initialization (`ip4_addr_t Wifi::ip = {0};`) to the top of Wifi.cpp.
Concerning threading issue: unfortunately I am not very familiar with that. Sure, here (at least in theory) the main thread and the other thread could access Wifi::ip simultaneously. In my application I am currently calling `get_local_ip()` from the main thread only - so, I am safe. But for general use it would be more safe to make it thread-safe. I never did such things. How it can be realized?

Reply:
We'd need to put a mutex-lock where we read and write it. I'm not very keen on adding a mutex for a single value, the overhead is so large. Let me think about it a while, we can put a warning comment on the get_local_ip function for now.

Reply:
I will do this during the next days (hopefully tomorrow evening).